lib/publiccloud/azure.pm

```diff
@@ -38,6 +48,21 @@ sub find_img {
     return;
 }
 
+sub find_img {
+    my ($self, $name) = @_;
+
+    $name = $self->filename_to_resgroup($name);
+    return unless $self->find_resgroup($name);
+
+    my $images = decode_json(script_output("az image list --resource-group '$name'"));
+    for my $i (@{$images}) {
```

Comment:
didn't get why you need a loop here ? 
`return $images[0]->{name}` would do the same as code in lines 58-60 .
Or you overlook if statement inside a loop ?

Reply:
No it's actually a left over, the --resource-group switch was added later.
Thanks for pointing that out!